force-app/main/default/triggers/ContactTrigger.trigger

```diff
@@ -19,10 +19,34 @@
 trigger ContactTrigger on Contact(before insert) {
 	// When a contact is inserted
 	// if DummyJSON_Id__c is null, generate a random number between 0 and 100 and set this as the contact's DummyJSON_Id__c value
+	if (trigger.isInsert && trigger.isBefore) {
+		for (Contact con : trigger.new) {
+			if (con.DummyJSON_Id__c == null) {
+				con.DummyJSON_Id__c = String.valueOf(Math.round(Math.random() * 100));				
+			}
+		}
+		
+	}
 
+	if (trigger.isInsert && trigger.isAfter){
 	//When a contact is inserted
 	// if DummyJSON_Id__c is less than or equal to 100, call the getDummyJSONUserFromId API
+		for(Contact con : trigger.new){
+			if(Integer.valueOf(con.DummyJson_Id__c) <= 100){
+				DummyJSONCallout.getDummyJSONUserFromId(con.DummyJson_Id__c);
```

Comment:
In a real scenario, you probably would not want to call a webservice within a loop. Or you want to at least make sure limit the amount it could be called. You can only call 50 @future methods per transaction. Are there other ways you can think of to bulkify this integration?